### ISSUE TITLE
NbInput - debounce

### ISSUE THREAD

**Issue by Tibing** [akveo/nebular#673]
### Issue type

**I'm submitting a ...**  (check one with "x")

* [ ] bug report
* [x] feature request

I think we have to add debounce property to the nbInput directive.

**Comment by Prefix1802** [comment]
@Tibing 

Could you elaborate why?

**Comment by Tibing** [comment]
Hi @Prefix1802, I mean, this is a really common use case when you need to add debouncing on your inputs: search, filter, maybe error handling, whatever. And if input itself can't debounce its events you have to write rxjs every time you need debouncing.
Anyway, it's just the proposal.

**Comment by Prefix1802** [comment]
@Tibing 

I understand now. You said property ;), so we can use it as a parameter.

+1

**Comment by fabioloreggian** [comment]
How would you like to use the debounce property?

My thought would be a `@Input() debounce: number;`
where this would be the debounce rate in milliseconds?

I am trying my hand at contributing to awesome angular projects and I really fancy nebular and ngx-admin.

**Comment by sulha199** [comment]
> How would you like to use the debounce property?
> 
> My thought would be a `@Input() debounce: number;`
> where this would be the debounce rate in milliseconds?
> 
> I am trying my hand at contributing to awesome angular projects and I really fancy nebular and ngx-admin.

Agree, and then use the debounce value to pipe the output such as on value change event.

@Tibing ,

Is this issue still open? I am interested to work on it?

**Comment by yassern** [comment]
and then provide a rxjs stream by @Output? 
but how we can access the input changes inside the nbInput directive?

**Comment by sethi-ishmeet** [comment]
@Prefix1802 @nnixaa @Tibing 

I would like to work on this feature request. I am just starting off to contributing on open source projects on GitHub. 

My Proposal:
- I would like to create a property `@Input() debounce: number;`.
- And after the debounced time, would like to emit the keyup event through an output property like:
```
  @Output()
  debouncedKeyup = new EventEmitter();
```

Please let me know if this is feasible.
